dimension/client.go

```diff
@@ -55,36 +56,76 @@ func (c *Client) Checker(ctx context.Context, check *healthcheck.CheckState) err
 	return c.hcCli.Checker(ctx, check)
 }
 
-// GetAreaTypes retrieves the Cantabular area-types associated with a dataset
-func (c *Client) GetAreaTypes(ctx context.Context, userAuthToken, serviceAuthToken, datasetID string) (GetAreaTypesResponse, error) {
-	logData := log.Data{
-		"method":     http.MethodGet,
-		"dataset_id": datasetID,
+func (c *Client) createDimensionGetRequest(ctx context.Context, userAuthToken, serviceAuthToken, urlPath string, logData log.Data, urlValues url.Values) (*http.Request, error) {
+	areasURL, err := c.baseURL.Parse(urlPath)
+	if err != nil {
+		return &http.Request{}, dperrors.New(
+			errors.Wrap(err, "failed to parse areas URL"),
+			http.StatusInternalServerError,
+			logData,
+		)
 	}
 
-	areaTypesURL, err := c.baseURL.Parse("area-types")
+	areasURL.RawQuery = urlValues.Encode()
+	reqURL := areasURL.String()
+
+	req, err := newRequest(ctx, http.MethodGet, reqURL, nil, userAuthToken, serviceAuthToken)
 	if err != nil {
-		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("failed to parse area-types URL: %w", err),
+		return &http.Request{}, dperrors.New(
+			errors.Wrap(err, "failed to create request"),
+			http.StatusBadRequest,
+			logData,
+		)
+	}
+	return req, nil
+}
+
+func checkDimensionGetResponse(logData log.Data, resp http.Response) error {
```

Comment:
No need to pass `logData` around in these functions. So long as you return logData embedded in the errors and errors are wrapped at each level the data for all errors will be unwrapped at the same time